app/components/tab_navigation_component.html.erb

```diff
@@ -1,14 +1,25 @@
 <%= content_tag(:nav, aria: { label: }, **tag_options, class: [*tag_options[:class], 'tab-navigation']) do %>
   <ul class="usa-button-group usa-button-group--segmented">
     <% routes.each do |route| %>
-      <li class="usa-button-group__item">
-        <%= render ButtonComponent.new(
-              url: route[:path],
-              big: true,
-              outline: !current_path?(route[:path]),
-              aria: { current: current_path?(route[:path]) ? 'page' : nil },
-            ).with_content(route[:text]) %>
-      </li>
+      <% if  route[:event_name] %>
+        <%=  render ClickObserverComponent.new(event_name: route[:event_name], class: "usa-button-group__item") do %>
```

Comment:
This approach works, but we dont get what we need from it, primarily, the request path, Not sure if we want to approach by adding that capability to the component.

Reply:
I feel like we could simplify this to logging a single event when clicking on tab navigation for the current path, so we wouldn't have to dynamically populate event name.

Something like...

```erb
<% if current_path?(route[:path) %>
  <%= render ClickObserverComponent.new(event_name: 'tab_navigation_current_page_clicked') %>
```

Reply:
I think that would work, but we wouldn't be able to pass in event name. I still think we would need the request path as a payload which the current implementation of this doesnt support. 

ex log 
"name": "tab_navigation_current_page_clicked",
  "properties": {
    "event_properties": {},
    "new_event": true,
    "path": "/api/logger",
    "service_provider": null,
    "session_duration": 1245.161211,
    "user_id": "anonymous-uuid",
    "locale": "en",
    "user_ip": "::1",
    "hostname": "localhost",
    "pid": 8930,
}

Reply:
Ah yeah, I see what you mean now. Yeah, I think we'd either want to find a way to support passing additional analytics attributes through the click component, or find a way to ensure that `path` is the actual page path where the click occurred and not just statically `/api/logger`.

Reply:
Take a look at my initial change or approach to see if it makes sense.

Reply:
Yeah I think the approach makes sense.

A few suggestions:

- I think `ClickObserverComponent` should accept `payload` as a constructor argument as a plain hash, so that it's a fully supported feature of the component and we don't rely on developers having to pass the correctly-formatted `data-` attribute
- I would suggest using something like `Hash#to_json` instead of manually creating a JSON object. In-fact, I think you pass a Ruby hash directly to a tag builder (`data: { payload: { path: '...' } }`) and it'll serialize as JSON automatically.